INSERT INTO `application`
(
`tag`,
`description`,
`is_accredited`,
`turnaround_time`,
`minimum_order`,
`sequencing_depth`,
`target_reads`,
`sample_amount`,
`sample_volume`,
`sample_concentration`,
`priority_processing`,
`details`,
`limitations`,
`percent_kth`,
`comment`,
`is_archived`,
`created_at`,
`is_external`,
`prep_category`
)
VALUES
(
'WGLLIFC030',
'Order of 49 or more samples. Whole genome sequencing,
30X coverage.',
1,
62,
49,
30,
650000000,
2000,
'25 - 50 uL', 
'8 - 250 ng/uL',
0,
'Whole-genome sequencing of large batches of samples (batch size of 49 or larger),
PE 2x150,
30x coverage using HiSeq X,
PCR-free library preparation. 75% of coverage guaranteed for samples fulfilling criteria. For cases where variant calling has been requested,
single nucleotide variants and short insertion/deletions will be identified using ensemble calling (GATK haplotype caller,
Freebayes,
samtools mpileup) and variants annotated using several sources of information; see MIP documentation for further details on annotation process.',
'Whole genome sequencing does not provide a complete coverage of the human genome due to certain regions being difficult to accurately represent in sequence data. Typical (minimum) coverage of the disease relevant gene panel is >99.5% (>97%) of bases covered at 10x or more. Using reference samples from National Institute of Standards and Technology in US,
we estimate sensitivity to >99.7% (>99.5%) for detection of SNVs and >85% (>90%) of indels. Individual samples may have a better or poorer performance,
which should be considered in the interpretation of the results.',
80,
'',
0,
'2019-05-17 14:00:00',
0,
'wgs'
);
